crates/forge/src/init.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::{bail, Result};
+
+use include_dir::{include_dir, Dir, DirEntry};
+use std::ffi::OsStr;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+static TEMPLATE: Dir = include_dir!("starknet_forge_template");
+
+fn create_with_template(dir: &Dir<'_>, base_path: &PathBuf, project_name: &str) -> Result<()> {
+    fs::create_dir_all(base_path)?;
+
+    for entry in dir.entries() {
+        let path = base_path.join(entry.path());
+
+        match entry {
+            DirEntry::Dir(d) => {
+                fs::create_dir_all(&path)?;
+                create_with_template(d, base_path, project_name)?;
+            }
+            DirEntry::File(f) => {
+                let contents = f.contents();
+                let contents = replace_project_name(contents, project_name);
+
+                fs::write(path, contents)?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn replace_project_name(contents: &[u8], project_name: &str) -> Vec<u8> {
+    // SAFETY: We control these files, and we know that they are UTF-8.
+    let contents = unsafe { std::str::from_utf8_unchecked(contents) };
```

Comment:
If we know that these are utf-8 why don't we just use `from_utf8` and unwrap/rethrow the error. I don't like the unsafe block here.

Reply:
copied from scarb, I can try change it